test/rtl/basic.mlir

```diff
@@ -3,43 +3,43 @@
 // CHECK-LABEL: func @test1(%arg0: i3) -> i50 {
 func @test1(%arg0: i3) -> i50 {
   // CHECK-NEXT:    %c42_i12 = rtl.constant(42 : i12) : i12
-  // CHECK-NEXT:    %0 = rtl.add %c42_i12, %c42_i12 : i12
-  // CHECK-NEXT:    %1 = rtl.mul %c42_i12, %0 : i12
+  // CHECK-NEXT:    %[[RES0:.*]] = rtl.add %c42_i12, %c42_i12 : i12
+  // CHECK-NEXT:    %[[RES1:.*]] = rtl.mul %c42_i12, %[[RES0:.*]] : i12
```

Comment:
`[[RES0:.*]]` is a capture expression. To use the captured value as a pattern, you need to use `[[RES0]]`.
Since the test only uses capture expressions, this is equivalent to just use wildcards as in:
```
// CHECK-NEXT:    %{{.*}} = rtl.mul %c42_i12, %{{.*}} : i12
```

Reply:
Oh, good catch I should have noticed that.  Thank you!

Reply:
Also, if you expect a number, you can be more specific and use `[[RES0:[0-9]+]]`. As a question of style, I would probably also include the % to the pattern: `[[RES0:%[0-9]+]]`.

Reply:
Thanks @jprotze!

Reply:
I'd personally just write `.*` because it is unambiguous and the pattern is simpler, but we're now so far down the line of personal preference that anything is fine :-)